lib/OptimizationMOI/src/OptimizationMOI.jl

```diff
@@ -238,12 +242,12 @@ end
 
 function _create_new_optimizer(model::MOI.AbstractOptimizer)
     if MOI.supports_incremental_interface(model)
-        return model
+        return MOI.instantiate(typeof(model))
```

Comment:
I don't think that this is guaranteed to work. If they give you an optimizer, use `MOI.empty!(model)` to empty it in preparation for a new solve.